src/streamlink/plugins/app17.py

```diff
@@ -28,28 +38,21 @@ def _get_streams(self):
         if page == 'user':
             res = http.get(self.url)
             userid = _userid_re.search(res.text).group(1)
-            data = {
-                "targetUserID": userid
-            }
-            api = http.post(API_URL, data=data)
-            info = re.sub(r'\\', '', api.text)
-            rid = _rid_re.search(info).group(1)
-            if rid == '0':
-                self.logger.info("Stream current unavailable.")
+            api = http.post(API_URL, data={"targetUserID": userid})
+            data = http.json(api, schema=_user_api_schema)
+            rid = data["liveStreamID"]
+            if rid == 0:
+                self.logger.info("Stream currently unavailable.")
                 return
 
             url = ROOM_URL.format(rid)
             res = http.get(url)
-        elif page == 'live':
+        else:
             res = http.get(self.url)
 
-        if res.status_code != 200:
-            self.logger.info("Not a valid room url.")
-            return
-
         status = _status_re.search(res.text).group(1)
```

Comment:
This will raise `AttributeError: 'NoneType' object has no attribute 'group'` if the page does not contain a stream. You could do something like: 
```python
status_m = _status_re.search(res.text)
if not status_m: 
    return 

if status_m.group(1) != 'true':
...
```